diff -ur aria2-1.37.0-ref/Makefile.am aria2-1.37.0/Makefile.am
--- aria2-1.37.0-ref/Makefile.am	2023-11-15 11:49:41.000000000 +0200
+++ aria2-1.37.0/Makefile.am	2023-12-02 17:18:03.324926252 +0200
@@ -1,4 +1,4 @@
-SUBDIRS =  po lib deps src doc test
+SUBDIRS =  po lib deps src
 
 ACLOCAL_AMFLAGS = -I m4 --install
 RST2HTML = @RST2HTML@
diff -ur aria2-1.37.0-ref/Makefile.in aria2-1.37.0/Makefile.in
--- aria2-1.37.0-ref/Makefile.in	2023-11-15 12:08:14.000000000 +0200
+++ aria2-1.37.0/Makefile.in	2023-12-02 17:18:03.324926252 +0200
@@ -451,7 +451,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = po lib deps src doc test
+SUBDIRS = po lib deps src
 ACLOCAL_AMFLAGS = -I m4 --install
 EXTRA_DIST = config.rpath \
 	     script-helper \
